operators/src/adapters/sparse_tiles_fill_adapter.rs

```diff
@@ -193,6 +263,89 @@ impl<T: Pixel> StateContainer<T> {
     fn current_idx_and_band_is_last_in_grid_run(&self) -> bool {
         self.current_idx == self.max_index() && self.current_band_idx == self.num_bands - 1
     }
+
+    fn set_current_time_from_initial_tile(&mut self, first_tile_time: TimeInterval) {
+        // if we know a bound we must use it to set the current time
+
+        let start_bound = self.time_bounds.start();
+        if start_bound < first_tile_time.start() {
+            log::debug!(
+                    "The initial tile starts ({}) after the start bound ({}), setting the current time to the start bound --> filling", first_tile_time.start(), start_bound
+                );
+            self.current_time = Some(TimeInterval::new_unchecked(
+                start_bound,
+                first_tile_time.start(),
+            ));
+            return;
+        }
+        if start_bound > first_tile_time.start() {
+            log::debug!(
+                    "The initial tile time start ({}) is before the exprected time bounds ({}). This means the data overflows the filler start bound.",
+                    first_tile_time.start(),
+                    start_bound
+                );
+        }
+        self.current_time = Some(first_tile_time);
+    }
+
+    fn set_current_time_from_bounds(&mut self) {
+        assert!(self.state == State::FillToEnd);
+        self.current_time = Some(TimeInterval::new_unchecked(
+            self.time_bounds.start(),
+            self.time_bounds.end(),
+        ));
+    }
+
+    fn update_current_time(&mut self, new_time: TimeInterval) {
+        if let Some(old_time) = self.current_time {
+            if old_time == new_time {
+                return;
+            }
+
+            debug_assert!(
+                old_time.end() <= new_time.start(),
+                "Time progress must be positive"
+            );
+
+            debug_assert!(
+                !old_time.is_instant() || old_time.end() < new_time.start(),
+                "Instant progress must be > 1"
+            );
+        }
+        self.current_time = Some(new_time);
+    }
+
+    fn current_time_fill_to_end_interval(&mut self) {
+        let current_time: TimeInterval = self
+            .current_time
+            .expect("current time must exist for fill to end state.");
+        debug_assert!(current_time.end() < self.time_bounds.end());
+
+        let new_time = if current_time.is_instant() {
+            TimeInterval::new_unchecked(current_time.end() + 1, self.time_bounds.end())
```

Comment:
wouldn't this create a gap in the output time series? if I have a dataset with only one raster image that is valid for an instant. it must fill the everything up to that point with nodata and everything from that point on with nodata. but right now it would create the nodata gap at the end one milli(?)second later and there is now a gap of one millisecond in the time series.

I think we should just disallow instants in time series. We have no use case where we need them anyway right?

Reply:
yes but it can't be the end of the element before since then it would have the same start. We have this case everywhere where time is modified.. We should remove the posibility of instants. However, this will require us to separate query and element Timeinterval. If we do not change the type it is still possible.